nanobenchmarks/main.cpp

```diff
@@ -164,6 +241,50 @@ void Main() {
   }
 }
 
+void Main() {
+  std::regex const filter(absl::GetFlag(FLAGS_benchmark_filter));
+  auto controller = PerformanceSettingsController::New();
+  std::unique_ptr<Logger> logger;
+  std::string const& filename = absl::GetFlag(FLAGS_log_to_mathematica);
+  if (!filename.empty()) {
+    logger = std::make_unique<Logger>(filename, /*make_unique=*/false);
+  }
+  std::println("{} {}",
+               principia::base::_cpuid::CPUVendorIdentificationString(),
+               principia::base::_cpuid::ProcessorBrandString());
+  std::println("Features: {}", principia::base::_cpuid::CPUFeatures());
+
+  TSCCalibration const calibration = CalibrateTSC(logger.get());
+  RunMatching<double, double>(filter,
+                              /*skip_if_empty=*/false,
+                              calibration,
+                              /*overhead_cycles=*/0,
+                              logger.get());
+  // NOTE: One block must be added for each pair of `Value, Argument` used in
+  // nanobenchmarks.
```

Comment:
Shouldn't we skip the calibration if none of the matchers of the relevant type match the filter?